### PR TITLE
Lattice no longer smooths with shuttle walls

### PR DESCRIPTION
**Pull request by Cyberboss**
:cl:
fix: Lattice no longer smooths with shuttle walls
/:cl:

### CODE REVIEW

**Comment by Cyberboss** [comment]
I'm an idiot

File: code/game/objects/structures/lattice.dm [hunk 1]
```diff
@@ -10,14 +10,13 @@
 	max_integrity = 50
 	layer = LATTICE_LAYER //under pipes
 	var/obj/item/stack/rods/stored
-	canSmoothWith = list(/obj/structure/lattice,
-	/turf/open/floor,
-	/turf/closed/wall,
-	/obj/structure/falsewall)
+	canSmoothWith = null
 	smooth = SMOOTH_MORE
```
Comment (ChangelingRain) [inline code/game/objects/structures/lattice.dm]: you'd have to make it SMOOTH_TRUE and have subtypes of everything minus shuttle walsl(which are just titanium walls)

File: code/game/objects/structures/lattice.dm [hunk 2]
```diff
@@ -10,14 +10,13 @@
...
-	/turf/open/floor,
-	/turf/closed/wall,
-	/obj/structure/falsewall)
-	smooth = SMOOTH_MORE
+	canSmoothWith = null
+	smooth = SMOOTH_TRUE
 	//	flags = CONDUCT
 
 /obj/structure/lattice/New()
+	canSmoothWith = list(/obj/structure/lattice,/turf/open/floor,/obj/structure/falsewall)
```
Comment (ChangelingRain) [inline code/game/objects/structures/lattice.dm]: you needs typesof for all of these minus false shuttle walls

Comment (Cyberboss) [inline code/game/objects/structures/lattice.dm]: ree

File: code/game/objects/structures/lattice.dm [hunk 3]
```diff
@@ -10,14 +10,12 @@
...
-	/turf/open/floor,
-	/turf/closed/wall,
-	/obj/structure/falsewall)
-	smooth = SMOOTH_MORE
+	canSmoothWith = null
+	smooth = SMOOTH_TRUE
 	//	flags = CONDUCT
 
 /obj/structure/lattice/New()
+	canSmoothWith = typesof(/obj/structure/lattice) + typesof(/turf/open/floor) + typesof(/obj/structure/falsewall) + typesof(/turf/closed/wall) - typesof(/turf/closed/wall/shuttle) - typesof(/obj/structure/falsewall/titanium)
```
Comment (ChangelingRain) [inline code/game/objects/structures/lattice.dm]: this still doesn't catch titanium walls, which is the actual path used for most shuttle walls

**Comment by ChangelingRain** [comment]
Really, this is a crude way of fixing the problem. Players can build titanium walls, and lattices should smooth or not smooth as appropriate.

**Comment by Cyberboss** [comment]
Reeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee

**Pull request closed without merging**